### ISSUE TITLE
Discussion on per-file license/header comment blocks

### ISSUE THREAD

**Issue by Anthchirp** [dials/dials#913]
Some files have headers such as
```python
#!/usr/bin/env python
#
# -*- Mode: Python; c-basic-offset: 2; indent-tabs-mode: nil; tab-width: 8 -*-
#
# $Id
#
# {filename}.py
#
#  Copyright (C) {year in the past} Diamond Light Source {or otherwise}
#
#  Author: {name}
#
#  This code is distributed under the BSD license, a copy of which is
#  included in the root directory of this package.
```

I tend to remove these, but as [@dagewa says](https://github.com/dials/dials/commit/edf059ef86e3ee1794349296d1f99104d6a2bbb9#commitcomment-34976534) I don't recall there being a recorded discussion on this.

So here is my reasoning:
1. In most cases the shebang (`#!`) does nothing as we mostly only run scripts from `/command_line`.
2. If you need to configure your editor (`-*- ...`) then you can do that outside of the source code file. Apart from that the settings are wrong. This is managed by black anyway.
3. We are not in SVN any more (`$Id`).
4. The license header is in the top license file, so unless this is different there is no signal here
5. The up-to-date author information is in `git blame`.

**Comment by graeme-winter** [comment]
The overall copyright should include everyone who has contributed - individual files are ~ useless - and unlikely to be copied on their own, so (for example with `xia2`) I have been removing this boiler plate. Almost always this is out of date anyways.

**Comment by rjgildea** [comment]
There are (were) also many that had the wrong filename or wrong author, which were obviously just copied and pasted from another file.

**Comment by ndevenish** [comment]
All of these points are 100% correct and I entirely applaud the removal of these chunks.

The only marker that should be up there is `# coding: utf-8` (or whatever the emacs version is if people prefer) and even that can be dropped once python 3.

Should we systematically remove them?

**Comment by dagewa** [comment]
I agree with the points above that the comment block headers are problematic and think that systematic removal make sense, rather than piecemeal deletion within other commits.

I should note though that previously STFC advised me to assert copyright and authorship like this:
```
__author__ = "David Waterman"
__copyright__= "Copyright 2019 United Kingdom Research and Innovation”
__license__ = "BSD 3-clause"
```
That was in the context of cctbx contributions, where as a minor contributor it was deemed I did not have enough control over top level assertions and should be rigorous about this in each new file (however, I don't think I have actually committed a whole new file to cctbx since getting that advice).

For DIALS, I think `__copyright__` and `__license__` per file are unnecessary, because these are properly set* at the top level and I would be involved in any future changes to these. However _I believe_ that STFC (UKRI) would prefer me to retain a claim of authorship over source code modules where I was the originator or have made a significant contribution. I appreciate this is a grey area and to date I have not been systematic about doing this in the code base.

However, it would be helpful if when these are removed, a change like this occurs:
```diff
- #  Author: David Waterman.
+ __author__ = "David Waterman"
```



*I have since noticed that the advice states
> the copyright statement should contain the latest year at which the code was developed (i.e. this year!)

so I ought to modify https://github.com/dials/dials/blob/master/LICENSE appropriately.

**Comment by ndevenish** [comment]
Right, I think that the `__copyright__` etc is intended for e.g. places hostile to proper attribution at the root level, and it's important to mark the file. Unless a file is entirely pasted in from an external source, with a different licence, I don't think we need that.

The authorship information for each file is embedded in the history for each file. I remain to be convinced that marking with `__author__` in a repository we control (outside of the root `__init__.py` import with other common programmatic metadata like `__version__`) is either beneficial or desired. It's just visual noise that everyone will ignore when reading the module.

I think my view is summarised well by a comment I found on a [stackoverflow](https://stackoverflow.com/a/1523456/1118662) discussing this:
> The next person to inherit the code has three choices: 1) update it all every time he/she edits the code 2) leave it alone, in which case it will be inaccurate 3) delete it all. Option 1 is a waste of their time, especially since they have absolutely zero confidence that the metadata was up to date when they received it. Options 2 and 3 mean that your time in putting it there in the first place was wasted. Solution: save everybody's time and don't put it there.

**Comment by dagewa** [comment]
I will check again with the IP & Licensing Manager regarding `__author__` asserts for the specific case of DIALS

**Comment by ndevenish** [comment]
Effectively fixed by 2a456212ea9bdabd8ef4f48920b73e31492cb03d
